lib/pg_audit_log/extensions/postgresql_adapter.rb

```diff
@@ -30,29 +30,44 @@ def rename_table(table_name, new_name)
     end
   end
 
-  def reconnect!
+  def reconnect!(...)
     super
     @last_user_id = @last_unique_name = nil
   end
 
-  def execute(sql, name = nil)
+  def execute(...)
     set_audit_user_id_and_name
-    super(sql, name = nil)
+    super
+  end
+
+  def internal_exec_query(...)
```

Comment:
This is new in Rails 7.1 and called under the hood by a handful of persistence methods (see below). Calling `set_audit_user_id_and_name` is idempotent (it only sets the user for the audit trigger function if the user is different from the currently-set user), so it seems better safe than sorry to monkey-patch this method too.